src/compas_timber/connections/half_lap.py

```diff
@@ -0,0 +1,152 @@
+                [1, 3, 2, 0],  # left
+                [3, 5, 4, 2],  # back
+                [5, 7, 6, 4],  # right
+                [7, 1, 0, 6],  # front
+            ],
+        )
+
+    @staticmethod
+    def _mesh_to_brep(negative_polyhedron_beam_a, negative_polyhedron_beam_b):
+        # Show Polyhedrons
+        negative_polyhedron_beam_a = Artist(negative_polyhedron_beam_a).draw()
+        negative_polyhedron_beam_b = Artist(negative_polyhedron_beam_b).draw()
+
+        # Breps from Polyhedrons
+        negative_brep_beam_a = RhinoBrep.CreateFromMesh(rs.coercemesh(negative_polyhedron_beam_a), True)
+        negative_brep_beam_b = RhinoBrep.CreateFromMesh(rs.coercemesh(negative_polyhedron_beam_b), True)
+        return negative_brep_beam_a, negative_brep_beam_b
+
+    def _create_negative_volumes(self):
+        # Get Planes from Beams
+        plane_a = Plane.from_frame(self.beam_a.faces[3])
+        plane_a_side1 = Plane.from_frame(self.beam_a.faces[0])
+        plane_a_side2 = Plane.from_frame(self.beam_a.faces[2])
+        plane_b1 = Plane.from_frame(self.beam_b.faces[1])
+        plane_b3 = Plane.from_frame(self.beam_b.faces[3])
+        plane_b_side1 = Plane.from_frame(self.beam_b.faces[0])
+        plane_b_side2 = Plane.from_frame(self.beam_b.faces[2])
+
+        # 4 Lines as Intersections between Beam Sides left/right, just like the Hashtag-Sign (#) that has 4 intersection points!
+        lines = []
+        x = intersection_plane_plane(plane_a_side1, plane_b_side1)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side1, plane_b_side2)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side2, plane_b_side2)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side2, plane_b_side1)
+        lines.append(Line(x[0], x[1]))
+
+        # Calculate Cut Direction of beam_b
+        vector_angle_a_b1 = angle_vectors(plane_a[1], plane_b1[1])
+        vector_angle_a_b3 = angle_vectors(plane_a[1], plane_b3[1])
+        if vector_angle_a_b1 > vector_angle_a_b3:
+            plane_b = plane_b1
+        else:
+            plane_b = plane_b3
+
+        # Get Cut Plane
+        plane_cut = self._cutplane()
+
+        # Create Polyhedrons
+        negative_polyhedron_beam_a = self._create_polyhedron(plane_a, plane_cut, lines)
+        negative_polyhedron_beam_b = self._create_polyhedron(plane_cut, plane_b, lines)
+
+        # Create BREP
+        return self._mesh_to_brep(negative_polyhedron_beam_a, negative_polyhedron_beam_b)
+
+    def restore_beams_from_keys(self, assemly):
+        """After de-serialization, resotres references to the main and cross beams saved in the assembly."""
+        self.beam_a = assemly.find_by_key(self.beam_a_key)
+        self.beam_b = assemly.find_by_key(self.beam_b_key)
+
+    def add_features(self):
+        negative_brep_beam_a, negative_brep_beam_b = self._mesh_to_brep()
```

Comment:
`self._mesh_to_brep()` is this working without arguments? shouldn't it call `self. _create_negative_volumes` ?